plugins/grafana-custom-plugins/grafana-dependency-plugin/src/DependencyPanel.tsx

```diff
@@ -44,26 +61,49 @@ export const DependencyPanel: React.FC<Props> = ({ options, data, width, height
 
   for (let i = 0; i < frame.length; i++) {
     const sourcePodName = sourcePodNames?.values.get(i);
+    const sourcePodLabel = sourcePodLabels?.values.get(i);
     const sourceNodeName = sourceNodeNames?.values.get(i);
     const destinationPodName = destinationPodNames?.values.get(i);
+    const destinationPodLabel = destinationPodLabels?.values.get(i);
     const destinationNodeName = destinationNodeNames?.values.get(i);
     const destinationServicePortName = destinationServicePortNames?.values.get(i);
     const octetDeltaCount = octetDeltaCounts?.values.get(i);
 
+    function getName(groupByLabel: boolean, source: boolean, labelJSON: string) {
+      if(!groupByLabel || labelJSON === undefined) {
+        return source ? sourcePodName : destinationPodName;
+      }
+      console.log('attempting to log JSON: '+labelJSON);
+      let labels = JSON.parse(labelJSON);
+      if(labels['app'] !== undefined) {
```

Comment:
In order to group by labels, I added support for the two basic ones I saw that had values for most pods. Should I make it check against a user-defined label?

Reply:
> the two basic ones I saw that had values for most pods

to clarify, where does the observation come from? which are the "most pods"? like I mentioned in the previous comment, is "app.kubernetes.io/name" less common than those two per your observation?

My point is, I'm not sure about which particular labels should we look for, so I doubt whether it's a good way to design this feature. I would like to take one step back to the motivation and design. For example, I see there are some common labels listed in k8s documentation: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/ , but I'm not convinced that we should look for these labels.

Reply:
<img width="494" alt="Screen Shot 2023-04-19 at 1 17 59 PM" src="https://user-images.githubusercontent.com/10016630/233190362-f82ca90d-cd1f-444a-8579-baf7f3c08488.png">

Reply:
<img width="496" alt="Screen Shot 2023-04-19 at 1 23 22 PM" src="https://user-images.githubusercontent.com/10016630/233191376-7cf471b8-5e3e-473a-8696-da25ab981066.png">

Reply:
- Could you also upload the screenshot of the diagram? 
- Does it mean user can only specify one label each time? Could we make use of the ad-hoc filter, showing all the available labels, and letting users to select one of more labels they would like to group by? I'm not sure if it is doable. If you would like to do a design review/have a design doc, we can discuss it in more details.

Reply:
I'll update with a screenshot of the diagram as well, but I don't think that an ad-hoc filter would be feasible for two reasons.
1. If pods are grouped by label, their pod label is used per square instead of the pod name itself. If multiple pod labels were used at once, I don't think there would be a good way of grouping them, especially if there are conflicts among labels. For example two pods could both be labelled by the same value for one label, but different values for another.
2. Ad-hoc filter can only check the database columns, anything more specific would require query variables, which cannot be included in the query if the query is to pass end to end testing. Dropdown menu is a great idea, I just don't know how it could be populated without a query variable. I can look into that, but if it doesn't pan out, users may need to manually enter.

Reply:
For the first one, I don't have a strong opinion on whether we should enable single label or multiple labels selection. If single label selection would make the graph representation easier, we can go with it.
 > For example two pods could both be labelled by the same value for one label, but different values for another.

Although I didn't get how does the above situation harm.

For the second one, we can remove this query from E2E test, if necessary. I think it worth exploring on query variables, ad-hoc filters, and maybe other Grafana configurations to see if it will make the dropdown menu work. I concern more on whether we can parse the Pod label string and split them into individual key value pairs.

Reply:
Yeah, splitting the label is more challenging than getting the rest of the query variable to work, since distinct values of only the key needs to be selected. There isn't a query that can parse the JSON and format output, so it may not be possible to include a dropdown menu. I hope that users wishing to group by label will already have some idea for which labels they wish to group by.
<img width="529" alt="Screen Shot 2023-04-26 at 10 19 38 AM" src="https://user-images.githubusercontent.com/10016630/234653473-bc09bf40-59a1-4801-bbd7-320c04c7ec30.png">

Reply:
why there are two source Pod groups with the same label value "web-client"?